panther_manager/include/panther_manager/plugins/action/dock_robot_action_node.hpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2024 Open Navigation LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PANTHER_MANAGER_DOCK_ROBOT_ACTION_NODE_HPP_
+#define PANTHER_MANAGER_DOCK_ROBOT_ACTION_NODE_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "behaviortree_ros2/bt_action_node.hpp"
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "opennav_docking_msgs/action/dock_robot.hpp"
+
+namespace panther_manager
+{
+
+class DockRobotAction : public BT::RosActionNode<opennav_docking_msgs::action::DockRobot>
+{
+  using Action = opennav_docking_msgs::action::DockRobot;
```

Comment:
I prefer `Action` because if you use this the namespace is added thanks to class `DockRobot::Action`

Reply:
Yep, I got your perspective, however:
- There is no point waiting with this alias declaration until a class is defined since u could use it in templating the base class `BT::RosActionNode<DockRobotAction>`
- This is message type, not a class DockRobot member (name should be self-explanatory, and not rely on the entity where is defined)
- In the whole project we follow convention <MessageName><Srv/Msg>

Consider the above aspects, but I'll leave the decision to u ;)

Reply:
Okay. That makes sense.